### ISSUE TITLE
PHP Errors in GlobalVariablesUnitTest.inc

### ISSUE THREAD

**Issue by connerbw** [WordPress-Coding-Standards/WordPress-Coding-Standards#948]
I accidentally opened  GlobalVariablesUnitTest.inc in PhpStorm and it got very angry at me.

---

```php
add_filter( 'comments_open', function( $open, $post_id ) {
	$post = get_post( $post_id ); // Ok, non-global variable in function scope.
	return 'page' === $page->post_type;
}, 10, 2 );
```

This will not work. There is no such variable `$page` in the aformetioned scope....

Source: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress/Tests/Variables/GlobalVariablesUnitTest.inc#L47

---

```php
// Allow overriding globals in functions within unit test classes.
// https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/300#issuecomment-158778606
trait WP_UnitTestCase {

	public function test_something() {
		global $tabs;
		$tabs = 50; // Ok.
	}
}

class Test_Class_A extends WP_UnitTestCase {

	public function test_something() {
		global $tabs;
		$tabs = 50; // Ok.
	}
	// ... snip ...        
```

PHP doesn't let you extend a trait... What is going on here?

Source: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress/Tests/Variables/GlobalVariablesUnitTest.inc#L62

---

Regards,

**Comment by jrfnl** [comment]
Unit tests are not working code. They just contain code testing that the sniff detects the correct things. The code in the unit test files is not parsed. just tokenized.
